src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -37,6 +37,16 @@
 REQUIRES_PYTHON_IDENTIFIER = cast(NormalizedName, "<Python from Requires-Python>")
 
 
+def has_ignore_dependencies(candidate: Candidate) -> bool:
+    ireq = candidate.get_install_requirement()
+    if ireq:
+        return ireq.has_ignore_dependencies
+    elif isinstance(candidate, AlreadyInstalledCandidate):
+        return candidate._ireq.has_ignore_dependencies
+    else:
+        return False
```

Comment:
Why not implement this on `Candidate` directly?

Reply:
should it be implemented for ExtrasCandidate and EditableCandidate?

Reply:
I think `_InstallRequirementBackedCandidate` should implement it as `return self._ireq.ignore_dependencies`, and everything else should `return False`? If I’m understanding the logic correctly.

Reply:
AlreadyInstalledCandidate should also implement `ignore_dependencies` not to install dependencies after re-using `pip install`

Reply:
Oh OK. Do what you need to do; the main thing here is to move the logic into Candidate subclasses.